packages/datadog-plugin-tedious/src/index.js

```diff
@@ -0,0 +1,135 @@
+'use strict'
+
+const Tags = require('../../../ext/tags')
+const Kinds = require('../../../ext/kinds')
+const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
+const tx = require('../../dd-trace/src/plugins/util/tx')
+
+const SQL_BATCH = 1
+
+function createWrapRequestClass (tracer) {
+  return function wrapRequestClass (Request) {
+    class RequestWithTrace extends Request {
+      constructor (sqlTextOrProcedure, callback) {
+        super(sqlTextOrProcedure, callback)
+        tracer.scope().bind(this)
+      }
+    }
+
+    return RequestWithTrace
+  }
+}
+
+function createWrapConnectionClass (tracer) {
+  return function wrapConnectionClass (Connection) {
+    class ConnectionWithTrace extends Connection {
+      constructor (config) {
+        super(config)
+        tracer.scope().bind(this)
+      }
+    }
+
+    return ConnectionWithTrace
+  }
+}
+
+function createWrapMakeRequest (tracer, config) {
+  return function wrapMakeRequest (makeRequest) {
+    return function makeRequestWithTrace (request, packetType) {
+      const connectionConfig = this.config
+      const scope = tracer.scope()
+      const childOf = scope.active()
+      const query = getQuery(request)
+
+      if (!query) {
+        return makeRequest.apply(this, arguments)
+      }
+
+      const span = tracer.startSpan(`tedious.request`, {
+        childOf,
+        tags: {
+          [Tags.SPAN_KIND]: Kinds.CLIENT,
+          'db.type': 'mssql',
+          'span.type': 'sql',
+          'component': 'tedious',
+          'service.name': config.service || `${tracer._service}-mssql`,
+          'resource.name': query
+        }
+      })
+
+      addConnectionTags(span, connectionConfig)
+      addDatabaseTags(span, connectionConfig)
+
+      if (packetType !== SQL_BATCH) {
+        addProcIdTags(span, request)
```

Comment:
How do we know when this is a query or a procedure? It looks like the check below in `getQuery` is different. Since this tag is not obfuscated, it's important to ensure it will never be a query. Maybe we can use a regular expression?

Reply:
The check below in `getQuery` is supposed to figure out what/where the SQL query is in the request. In the case of `SQL_BATCH`-type requests, we know there is a SQL query (because of `getQuery`), but that it's also not a procedure call. `SQL_BATCH`-type requests are currently the only type of requests that have a query but are also not procedure calls.

I can see the issue though if the library were to update and start supporting more request types that were implemented differently. I can make it so that the `request.sqlTestOrProcedure` must be of the form `sp_[a-z]+`.

Reply:
While a bit more complicated, I think it would be worth it since there are security implications.